tinyiiod.c

```diff
@@ -134,41 +146,39 @@ void tinyiiod_write_xml(struct tinyiiod *iiod)
 void tinyiiod_do_read_attr(struct tinyiiod *iiod, const char *device,
 			   const char *channel, bool ch_out, const char *attr, bool debug)
 {
-	char buf[128];
 	ssize_t ret;
 
 	if (channel)
 		ret = iiod->ops->ch_read_attr(device, channel,
-					      ch_out, attr, buf, sizeof(buf));
+					      ch_out, attr, iiod_buf[READ_BUFFER], IIOD_BUFFER_SIZE);
```

Comment:
I could accept that UART transmission are hard to section and read multiple times in chunks of 128, but i do not understand the need for two 1000 bytes buffers. If you need that much memory try to optimize and use only one.

Reply:
Also for UART read/writes a ring buffer is a nice idea and would probably decrease that 1000 bytes size. Just a thought.